src/main/java/de/qabel/desktop/ui/accounting/GsonIdentity.java

```diff
@@ -0,0 +1,43 @@
+package de.qabel.desktop.ui.accounting;
+
+import com.google.gson.annotations.Expose;
+import java.util.HashMap;
+
+
+public class GsonIdentity extends GsonEntity {
+
+	@Expose
+	private HashMap<String, byte[]> keys = new HashMap<>();
+
+	private  byte[] publicKey;
+	private byte[] privateKey;
+
+
+	public byte[] getPublicKey() {
+		return publicKey;
+	}
+
+	public void setPublicKey(byte[] publicKey) {
+		this.publicKey = publicKey;
+	}
+
+	public byte[] getPrivateKey() {
+		return privateKey;
+	}
+
+	public void setPrivateKey(byte[] privateKey) {
+		this.privateKey = privateKey;
+	}
+
+	public void generateKeyStructure() {
```

Comment:
if `generateKeyStructure` and `buildKeyPair` would be called on get- / setPrivateKey, the caller wouldn't need stateful knowledge (and the methods could then be private)